src/playbin.js

```diff
@@ -102,29 +108,43 @@ var Playbin = GObject.registerClass({
         if (width_ok && height_ok) {
             this._video_width = width;
             this._video_height = height;
-            this._video_widget.show_now();
+            this._updateCropArea(this.get_allocation());
         }
     }
 
-    _playbackDone () {
+    _onEndOfStream () {
         this.emit('done');
         this._playbin.set_state(Gst.State.NULL);
         this.remove(this._video_widget);
-        this.uri = null;
+        this._uri = null;
         this._playbin = null;
         this._videocrop = null;
         this._video_widget = null;
+        this._video_width = 0;
+        this._video_height = 0;
+    }
+
+    _onStateChanged (msg) {
+        if (this._started)
+            return;
+
+        const [, newstate] = msg.parse_state_changed();
+
+        if (newstate === Gst.State.PLAYING) {
+            this.set_visible_child(this._video_widget);
+            this.get_style_context().remove_class('locked');
```

Comment:
I'd rather you set `locked = false` directly here, so there are not two places where style classes are tweaked... But also I would consider emitting a signal from here and setting it from the lockscreen through `_updateUI()` instead.